src/utils.rs

```diff
@@ -0,0 +1,109 @@
+//! Utils used in the crate.
+
+use crate::filesystem::FatFileSystem;
+use core::ops::{BitAnd, Not};
+use num_traits::Num;
+use storage_device::StorageDevice;
+
+/// Align the address to the next alignment.
+///
+/// The given number should be a power of two to get coherent results!
+///
+/// # Panics
+///
+/// Panics on underflow if align is 0.
+/// Panics on overflow if the expression `addr + (align - 1)` overflows.
+pub fn align_up<T: Num + Not<Output = T> + BitAnd<Output = T> + Copy>(addr: T, align: T) -> T {
+    align_down(addr + (align - T::one()), align)
+}
+
+/// Align the address to the previous alignment.
+///
+/// The given number should be a power of two to get coherent results!
+///
+/// # Panics
+///
+/// Panics on underflow if align is 0.
+pub fn align_down<T: Num + Not<Output = T> + BitAnd<Output = T> + Copy>(addr: T, align: T) -> T {
+    addr & !(align - T::one())
+}
+
+/// Retrieve the parent of a given path.
+pub fn get_parent(path: &str) -> (&str, &str) {
+    let separator_index_opt = path.rfind('/');
+
+    if let Some(separator_index) = separator_index_opt {
+        let (first, last) = path.split_at(separator_index);
+        (first, &last[1..])
+    } else {
+        ("", path)
+    }
+}
+
+/// Permite to split a path.
```

Comment:
Add the same tests, and update the doc

```suggestion
/// Splits a path at the first `/` it encounters.
///
/// Returns a tuple of the parts before and after the cut.
///
/// ```
/// let my_path = "/top_level_dir/middle_dir/sub_dir";
/// let (before, after) = split_path(my_path);
/// assert!(before == "/top_level_dir");
/// assert!(after.unwrap() == "middle_dir/sub_dir");
/// ```
///
/// If no `/` is encountered, the "after" part is None:
///
/// ```
/// let my_path = "/top_level_dir";
/// let (before, after) = split_path(my_path);
/// assert!(before == "/top_level_dir");
/// assert!(after.is_none());
/// ```
///
/// # Panics
/// 
/// @Thog forgot to update the panic conditions
```